wideq/client.py

```diff
@@ -349,8 +353,27 @@ def enum_name(self, key, value):
         """Look up the friendly enum name for an encoded value.
         """
         options = self.value(key).options
+        if value not in options:
+            logging.warning(
+                'Value `%s` for key `%s` not in options: %s. Values from API: '
+                '%s', value, key, options, self.data['Value'][key]['option'])
+            return _UNKNOWN
         return options[value]
 
+    def reference_name(self, key: str, value: Any) -> str:
+        """Look up the friendly name for an encoded reference value.
+
+        :param key: The referenced key.
+        :param value: The value whose name we want to look up.
+        :returns: The friendly name for the referenced value.  If no name
+            can be found `-` will be returned.
+        """
+        value = str(value)
+        reference = self.value(key).reference
+        if value in reference:
+            return reference[value]['_comment']
+        return '-'
```

Comment:
Maybe you can help me out with one small gap in my understanding: is it important that the "not found" return value here be `'-'`? Would it work just as well for it to be `None` instead (and the return type for this function to be `Optional[str]`)? If so, that would be a tad more Pythonic and self-documenting. But it seems like there might be an API-specific specific meaning to `'-'` that I'm missing.

Reply:
No, there isn't any particular reason, I changed it to return None instead.  (especially because in the DryerStatus class we return 'Off' if the values is '-'.  So None makes more sense here.